server/player/src/packet_handlers/inventory.rs

```diff
@@ -406,12 +406,83 @@ fn handle_single_click(
 }
 
 fn handle_double_click(
-    _game: &mut Game,
-    _world: &mut World,
-    _player: Entity,
+    game: &mut Game,
+    world: &mut World,
+    player: Entity,
     _packet: ClickWindow,
 ) -> anyhow::Result<()> {
-    // not sure what to do here?
+    // Double click gathers items into the picked stack until it hits the max stack size, or it has picked up all possible items
+
+    // Keep track of what slots are modified so the client can be informed
+    let mut modified_slots: smallvec::SmallVec<[SlotIndex; 2]> = smallvec![];
+
+    let stack_size;
+
+    let new_picked_count = {
```

Comment:
I create a new scope for this part of the handler to cleanly handle dropping all the stuff I use before I update the PickedItem